Replace log4rs with tracing

amy-keibler opened the pull request:
[`log4rs`](https://crates.io/crates/log4rs) has a transitive dependency on `traitobject` that has a vulnerability. There hasn't been a release of `log4rs` in 11 months, so the vulnerability will not likely be addressed in the near future.

[`tracing`](https://crates.io/crates/tracing) is a logging/tracing framework from the Tokio project that is compatible with the output of the [`log`](https://crates.io/crates/log) crate, but provides additional constructs such as spans to make debugging easier compared to correlating lines of output.

This change should be approximately equivalent to the previous solution. An added benefit is that five verbose arguments (`-vvvvv`) cause us to read the `RUST_LOG` environment variable for a tracing directive ([documentation](https://docs.rs/tracing-subscriber/0.3.1/tracing_subscriber/filter/struct.EnvFilter.html#directives)). This directive can be used to target a specific subsection of the program or a dependency, without turning up the logging level for everything (e.g. `RUST_LOG=debug,reqwest=trace` to investigate a network issue).

Additionally, this change includes some code cleanup changes such as removing the old crate macro import syntax for `serde` and removing a ternary macro.

Fixes #59

amy-keibler committed:
Replace log4rs with tracing

log4rs has a transitive dependency on traitobject that has a vulnerability.
There hasn't been a release of log4rs in 11 months, so the vulnerability will
not likely be addressed in the near future.

tracing is a logging/tracing framework from the Tokio project that is compatible
with the output of the log crate, but provides additional constructs such as
spans to make debugging easier compared to correlating lines of output.

This change should be approximately equivalent to the previous solution. An
added benefit is that five verbose arguments ("-vvvvv") cause us to read the
RUST_LOG environment variable for a tracing directive (documented:
https://docs.rs/tracing-subscriber/0.3.1/tracing_subscriber/filter/struct.EnvFilter.html#directives).
This directive can be used to target a specific subsection of the program or a
dependency, without turning up the logging level for everything (e.g.
"RUST_LOG=debug,reqwest=trace" to investigate a network issue).

Additionally, this change includes some code cleanup changes such as removing
the old crate macro import syntax for serde and removing a ternary macro.

Signed-off-by: Amelia Keibler <akeibler@sonatype.com>

amy-keibler commented:
Yay, finally passing CI ✅

bhamail reviewed (approved):
Works great for me! 

(well, works great after I remember to add valid IQ credentials. Doh! Created a separate issue to track that sort of thang.)

One other thought: Should we "empty":
```
{
  "ignore": [
  ]
}
```
 or "remove" the the `.pants-ignore` file, because we no longer have to ignore that vulnerability?

amy-keibler committed:
Remove vulerability from the ignore file

Since we update the library, we are no longer vulnerable to this
particular issue.

Signed-off-by: Amelia Keibler <akeibler@sonatype.com>

amy-keibler commented:
Good question! I think leaving the empty `.pants-ignore` `ignore` list is probably best, so I added that to these changes.
